app/commands/user/reputation_token/award_for_issue.rb

```diff
@@ -0,0 +1,62 @@
+class User
+  class ReputationToken
+    class AwardForIssue
+      include Mandate
+
+      initialize_with :params
+
+      def call
+        return unless label_change?
+        return unless author
+        return unless author.maintainer? || author.admin?
+
+        if reputation_level
+          create_token!
+        else
+          delete_token!
+        end
+      end
+
+      private
+      def label_change?
+        %w[labeled unlabeled].include?(params[:action])
+      end
+
+      memoize
+      def reputation_level
+        # Sort descendingly to award greatest possible reputation
+        %i[massive large].find do |type|
+          params[:labels].include?(Github::IssueLabel.for_type(:size, type))
+        end
+      end
+
+      memoize
+      def author
+        User.find_by(github_username: params[:opened_by_username])
+      end
+
+      def create_token!
+        User::ReputationToken::Create.(
+          author,
+          :issue_author,
+          level: reputation_level,
+          repo: params[:repo],
+          issue_node_id: params[:node_id],
+          issue_number: params[:number],
+          issue_title: params[:title],
+          external_url: params[:html_url],
+          opened_at: params[:opened_at]
+        )&.tap do |token|
+          token.update!(level: reputation_level)
+        end
+      end
+
+      def delete_token!
+        User::ReputationTokens::IssueAuthorToken.where(user: author, external_url: params[:html_url]).first&.tap do |token|
```

Comment:
Ideally, I'd be using the uniqueness key here to find the token, but for that I need to create the token first. Any thoughts here? There is no index on `external_url` either, but it might not be that bad with the filter already using the type of the token and the user.

Reply:
You can use `find_by` instead of `where...first`

Reply:
I think it'll be fine with the user/type filter. Run a query in a rails c and check how long it takes and if it's slow send me the query to check?

Reply:
I'm ok with this not being a command for now, but if we get to the point we're deleting other tokens, it should be DRY'd into that as it's a two-step thing.

Reply:
I'm fine with adding it as a command.

Reply:
I've updated